clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/ValidateSchemaAndBuildDictOutputFormat.java

```diff
@@ -78,13 +79,17 @@ private static void createAndSetDirectoryPermission(
   protected static void setValidateSchemaAndBuildDictionaryOutputDirPath(JobConf job) throws IOException {
     // parent directory
     String parentOutputDir = job.get(MAPPER_OUTPUT_DIRECTORY);
-    Path outputPath = new Path(parentOutputDir);
-    FileSystem fs = outputPath.getFileSystem(job);
-    createAndSetDirectoryPermission(fs, outputPath, "777");
+    if (!parentOutputDir.equalsIgnoreCase(VALIDATE_SCHEMA_AND_BUILD_DICTIONARY_MAPPER_OUTPUT_PARENT_DIR_DEFAULT)) {
+      // '/tmp' is default parent folder
+      Path outputPath = new Path(parentOutputDir);
+      FileSystem fs = outputPath.getFileSystem(job);
+      createAndSetDirectoryPermission(fs, outputPath, "777");
```

Comment:
`/tmp` would generally not be owned by `VenicePushJob`. In `createAndSetDirectoryPermission`, we try to set the permission of `/tmp` which will fail if the sticky bit is set on `/tmp`. This is likely to be set since `/tmp` is a common directory for everyone that uses that HDFS deployment.

So, this could lead to other issues. Can we remove the parent directory creation and leave it up to the operator to set up before any VPJ execution?

Reply:
I think I know why the ACL issues are happening in VPJ

The stacktrace has this:
```at com.linkedin.venice.hadoop.ValidateSchemaAndBuildDictOutputFormat.createAndSetDirectoryPermission(ValidateSchemaAndBuildDictOutputFormat.java:66)```

This corresponds to setting `777` permissions on `/tmp/venice`. Other headless accounts are trying to change permissions on `/tmp/venice` where only the owner is allowed to do so.

We can set permissions only at the time of directory creation. We need not try to set 777 every time

Reply:
Yes, this change made the diff cleaner. Thanks for pointing it out.